tests/testing.py

```diff
@@ -0,0 +1,312 @@
+
+
+class Base(unittest.TestCase):
+    delegator_wallet = None
+    delegator_address = None
+
+    validator_wallet = None
+    validator_address = None
+    validator_operator_address = None
+
+    ledger_client = None
+    db = None
+    db_cursor = None
+    gql_client = None
+
+    @classmethod
+    def setUpClass(cls):
+        validator_mnemonic = "nut grocery slice visit barrel peanut tumble patch slim logic install evidence fiction shield rich brown around arrest fresh position animal butter forget cost"
+        validator_seed_bytes = Bip39SeedGenerator(validator_mnemonic).Generate()
+        validator_bip44_def_ctx = Bip44.FromSeed(validator_seed_bytes, Bip44Coins.COSMOS).DeriveDefaultPath()
+        cls.validator_wallet = LocalWallet(PrivateKey(validator_bip44_def_ctx.PrivateKey().Raw().ToBytes()))
+        cls.validator_address = str(cls.validator_wallet.address())
+        cls.validator_operator_address = Address(bytes(cls.validator_wallet.address()), prefix="fetchvaloper")
+
+        delegator_mnemonic = "dismiss domain uniform image cute buzz ride anxiety nose canvas ripple stock buffalo bitter spirit maximum tone inner couch forum equal usage state scan"
+        delegator_seed_bytes = Bip39SeedGenerator(delegator_mnemonic).Generate()
+        delegator_bip44_def_ctx = Bip44.FromSeed(delegator_seed_bytes, Bip44Coins.COSMOS).DeriveDefaultPath()
+        cls.delegator_wallet = LocalWallet(PrivateKey(delegator_bip44_def_ctx.PrivateKey().Raw().ToBytes()))
+        cls.delegator_address = str(cls.delegator_wallet.address())
+
+        cfg = NetworkConfig(
+            chain_id="testing",
+            url="grpc+http://localhost:9090",
+            fee_minimum_gas_price=1,
+            fee_denomination="atestfet",
+            staking_denomination="atestfet",
+        )
+
+        gov_cfg = grpc.insecure_channel('localhost:9090')
+
+        cls.ledger_client = LedgerClient(cfg)
+        cls.gov_module = query_pb2_grpc.QueryStub(gov_cfg)
+
+        transport = AIOHTTPTransport(url="http://localhost:3000")
+        cls.gql_client = Client(transport=transport, fetch_schema_from_transport=True)
+
+        cls.db = psycopg.connect(
+            host="localhost",
+            port="5432",
+            dbname="postgres",
+            user="postgres",
+            password="postgres",
+            options=f'-c search_path=app'
+        )
+
+        cls.db_cursor = cls.db.cursor()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.db.close()
+
+
+class GovernanceTests(Base):
+    vote_tx = None
+    denom = "atestfet"
+    amount = "10000000"
+    option = "YES"
+
+    def setUp(self):
+        self.msg = gov_tx.MsgVote(
+            proposal_id=1,
+            voter=self.validator_address,
+            option=gov_pb2.VoteOption.VOTE_OPTION_YES
+        )
+        self.vote_tx = Transaction()
+        self.vote_tx.add_message(self.msg)
+
+    def testGovernanceProposal(self):
```

Comment:
This could probably be combined with `setUp`. I think we can still make assertions in there.